To include support for BZIP2 compression in GunPG for W32, the patch
below should be applied to a stock bzip2-1.0.2 source.  The Build as
usual using the mingw32 cross compiler package from Debian and install
the library and header file on top of the cross compiler installation
(/usr/i586-mingw32msvc/lib/).  Note that for ease of maintenance we
don't use a DLL.                                      [wk 2005-03-14]


diff -u orig/bzip2-1.0.2/Makefile bzip2-1.0.2/Makefile
--- orig/bzip2-1.0.2/Makefile	2002-01-26 00:34:53.000000000 +0100
+++ bzip2-1.0.2/Makefile	2004-11-03 14:10:45.000000000 +0100
@@ -2,9 +2,9 @@
 SHELL=/bin/sh
 
 # To assist in cross-compiling
-CC=gcc
-AR=ar
-RANLIB=ranlib
+CC=i586-mingw32msvc-gcc
+AR=i586-mingw32msvc-ar
+RANLIB=i586-mingw32msvc-ranlib
 LDFLAGS=
 
 # Suitably paranoid flags to avoid bugs in gcc-2.7
diff -u orig/bzip2-1.0.2/bzlib.h bzip2-1.0.2/bzlib.h
--- orig/bzip2-1.0.2/bzlib.h	2001-12-30 03:19:45.000000000 +0100
+++ bzip2-1.0.2/bzlib.h	2004-11-03 14:32:41.000000000 +0100
@@ -113,7 +114,7 @@
 /* Need a definitition for FILE */
 #include <stdio.h>
 
-#ifdef _WIN32
+#if defined( _WIN32 ) && 0
 #   include <windows.h>
 #   ifdef small
       /* windows.h define small to char */



